### PR TITLE
Various content improvements (Components / State sections)

### PR DESCRIPTION
**Pull request by jackie-pc**
Also links to next/prev page will show more context if needed.

E.g. "State Overview" instead of just "Overview".

### CODE REVIEW

File: pcweb/pages/docs/components/overview.py [hunk 1]
```diff
@@ -15,10 +15,12 @@
 code_example1 = """rx.text('Hello World!', color='blue', font_size="1.5em")"""
 code_example2 = """
 rx.hstack(
+    # Static 50% progress
```
Comment (jackie-pc) [inline pcweb/pages/docs/components/overview.py:18]: Comment to spell out what is happening to reader.

File: pcweb/pages/docs/components/overview.py [hunk 2]
```diff
@@ -31,11 +33,11 @@
     definition(
         "Children",
         rx.unordered_list(
-            rx.list_item("Other Reflex components nested inside a component."),
+            rx.list_item("Text or other Reflex components nested inside a component."),
```
Comment (jackie-pc) [inline pcweb/pages/docs/components/overview.py:36]: The example immediately after this shows that a regular string can be passed too.

File: pcweb/pages/docs/components/overview.py [hunk 3]
```diff
@@ -69,13 +71,13 @@ def components_overview():
             They let you split the UI into independent, reusable pieces, and let you think about each piece in isolation."""
         ),
         doctext(
-            "If you are familiar with React, Reflex components are simply wrappers around React components. ",
+            "Reflex components wrap React components behind the scenes, enabling a pure Python development experience.",
```
Comment (jackie-pc) [inline pcweb/pages/docs/components/overview.py:74]: Ensuring that this "wrapping" is framed positively - hint at why we do this.

File: pcweb/pages/docs/components/overview.py [hunk 4]
```diff
@@ -88,10 +90,15 @@ def components_overview():
             ),
         ),
         rx.tablet_and_desktop(
-            rx.hstack(
-                *definitions,
+            rx.grid(
```
Comment (jackie-pc) [inline pcweb/pages/docs/components/overview.py:93]: When I changed the definitions, the two boxes were no longer symmetrically sized in the hstack.... Using grid to make the equal split happen (not sure if it's the best way!)

Comment (Alek99) [reply pcweb/pages/docs/components/overview.py:93]: This way is fine!

**Review by Alek99 — APPROVED** [review]
lgtm

File: pcweb/pages/docs/state/utility_methods.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        rx.text(State.cookies),
+    )
+
+
+app = rx.App(state=State)
+app.add_page(index)
+"""
+
+@docpage()
+def utility_methods():
```
Comment (picklelo) [inline pcweb/pages/docs/state/utility_methods.py:33]: Good idea separating this out to its own section

File: pcweb/pages/docs/state/utility_methods.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+            ),
+            rx.list_item(
+                rx.text(rx.code("get_cookies()"), ": Returns the current user's browser cookies."),
+            ),
+            padding_bottom="0.7em"
+        ),
+        doccode(code1),
+        doctext("Check out the ",
+                doclink("State class definition",
+                        href="https://github.com/reflex-dev/reflex/blob/main/reflex/state.py"),
```
Comment (picklelo) [inline pcweb/pages/docs/state/utility_methods.py:59]: We should link to this page instead: https://reflex.dev/docs/api-reference/state (can do in a follow up)

**Review by picklelo — APPROVED** [review]

**Pull request merged**